fixes #919 Polling on subscriber socket recvfd seems broken

bsirang opened the pull request:
`sub0_recv_cb()` was not calling `nni_pollable_raise()` on `sock->recvable`.

bsirang commented:
@gdamore Please review. I don't understand the internals of nng well enough to know if the `nni_pollable_raise()` call needs to be placed elsewhere, but this seems to have solved the issue I encountered.

gdamore commented on src/protocol/pubsub0/sub.c:
```diff
@@ -414,7 +414,7 @@ sub0_recv_cb(void *arg)
 		nni_msg_free(msg);
 	}
-
+	nni_pollable_raise(sock->recvable);
```
This isn't quite right, but it's close.  The issue here is that we don't want to raise this FD unless we passed the SUB filter.  So what needs to happen is up around line 382 we need to set a flag about whether this was matched or not, and then only do the raise down here if we matched.  Otherwise we may wind up doing a lot more spurious wakeups.

bsirang commented:
@gdamore Please review again.

Note that I added the flag below this check as well
```
		if (ctx == nni_list_last(&sock->ctxs)) {
			dup = msg;
			msg = NULL;
		} else if (nni_msg_dup(&dup, msg) != 0) {
			continue; // TODO: Bump a stat!
		}
```

Because if `nni_msg_dup()` fails, it seemed like we also wouldn't want to raise the pollable fd. I updated the unit test to test both cases (subscriber topic matches vs. doesn't match)
